llarp/CMakeLists.txt

```diff
@@ -1,5 +1,21 @@
 include(Version)
 
+target_sources(lokinet-cryptography PRIVATE
+  crypto/crypto_libsodium.cpp
+  crypto/crypto.cpp
+  crypto/encrypted_frame.cpp
+  crypto/types.cpp
+)
+
+# libcrypt interface library to optionally use libcrypt if we can find it
+add_library(lokinet-libcrypt INTERFACE)
+pkg_check_modules(CRYPT libcrypt IMPORTED_TARGET)
+if(CRYPT_FOUND AND NOT CMAKE_CROSSCOMPILING)
+  add_definitions(-DHAVE_CRYPT)
+  target_link_libraries(lokinet-libcrypt INTERFACE PkgConfig::CRYPT)
+  message(STATUS "using libcrypt ${CRYPT_VERSION}")
+endif()
```

Comment:
I feel like this belongs in `external/`

Reply:
maybe, it is an optional system dep rather than a submodule so i didnt put it in external